hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java

```diff
@@ -1815,6 +1815,28 @@ private int getWhenNoSSLDetected() {
     public static final HazelcastProperty PERSISTENCE_AUTO_CLUSTER_STATE_STRATEGY = new HazelcastProperty(
             "hazelcast.persistence.auto.cluster.state.strategy", ClusterState.NO_MIGRATION);
 
+    /**
+     * Enables/disables Hazelcast Alto. If present, overrides the enabled
+     * field in AltoConfig.
+     *
+     * @see com.hazelcast.config.alto.AltoConfig
+     * @since 5.3
+     */
+    @Beta
+    public static final HazelcastProperty ALTO_ENABLED = new HazelcastProperty(
+            "hazelcast.alto.enabled");
+
+    /**
+     * Sets the eventloop thread count in Hazelcast Alto. If present,
+     * overrides the eventloopCount field in AltoConfig.
+     *
+     * @see com.hazelcast.config.alto.AltoConfig
+     * @since 5.3
+     */
+    @Beta
+    public static final HazelcastProperty ALTO_EVENTLOOP_COUNT = new HazelcastProperty(
```

Comment:
I think having two things (a property here and a config element in AltoConfig) for the same config option is a bit confusing. What if we don't define them here, but take them into consideration from system properties in TpcServerBootstrap? It will essentially behave the same, but at least we won't define an extra property and make it configurable in two ways using the Hazelcast configuration

Reply:
What you are asking is whether we should make this hazelcast property public or not.

cc @pveentjer

Reply:
Then I'm in favor of HazelcastProperty having preference above the alto config.

Reply:
Metting is right: the duplication of this property causes problems since there are 2 paths of configuration. Ok, drop the ClusterProperty but do check the system properties and the configured HZ properties. Yuk...

Reply:
https://github.com/hazelcast/hazelcast/pull/23319/commits/3961f91393cdf19f4e0562e322cae674ef04a4a5